### PR TITLE
[Wiz] Filter issues by status

### PR DESCRIPTION
**Pull request by phalbert**
# Description

This PR introduces the ability to filter issues by their status and includes a hard limit of 50,000. Previously, the system was restricted to displaying only the top 250 recent issues. 

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [x] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [x] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
lets add tests, look at fake integration

**Comment by phalbert** [comment]
> lets add tests, look at fake integration

@Tankilevitch, This was taking long to figure out. Let me first familiarise myself with them while doing the Jenkins build stages. I say we release this as I have tested using the check list

File: integrations/wiz/main.py [hunk 1]
```diff
@@ -26,31 +29,49 @@ def init_client() -> WizClient:
...
+
+    async for issues in wiz_client.get_issues(status_list):
+        yield issues
+
+
+@ocean.on_resync(ObjectKind.CONTROL)
+async def resync_controls(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    wiz_client = init_client()
+
+    async for _issues in wiz_client.get_cached_issues():
```
Comment (Tankilevitch) [inline integrations/wiz/main.py:59]: this will not work if user didn't choose to map kind `issue`, lets not cache

Comment (phalbert) [reply integrations/wiz/main.py]: I have changed to the default issues method with default status list of ["OPEN", "IN_PROGRESS"]

File: integrations/wiz/CHANGELOG.md [hunk 2]
```diff
@@ -7,471 +7,360 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 <!-- towncrier release notes start -->
 
-## 0.1.57 (2024-09-12)
+## 0.1.58 (2024-09-12)
```
Comment (Tankilevitch) [inline integrations/wiz/CHANGELOG.md:11]: Missing Fixes title

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**